sodetlib/operations/translate_tes_parameters.py

```diff
@@ -0,0 +1,569 @@
+import numpy as np
+from scipy.interpolate import interp1d
+from scipy.optimize import curve_fit
+from sodetlib.operations.iv import IVAnalysis
+from sodetlib.operations.bias_steps import BiasStepAnalysis
+from tqdm.auto import trange
+
+
+import warnings
+warnings.filterwarnings('ignore')
+np.seterr(all="ignore")
+
+
+def translate_tes_params(iv_file, bs_file, save_iv=False, save_bs=False):
+    """
+    Driver function that recalculates IV TES parameters using the
+    thevenin equivalent voltage. Then computes the RP curve following
+    Satoru's methods. Finally, load in and recalculate bias step analysis
+    TES parameters using a shifted RP curve determined by the change in 
+    optical power between operations. 
+    
+    Args:
+    -----
+    iv_file : str
+        filepath to iv_analysis.npy file to re-analyze.
+        
+    bs_file : str
+        filepath to bs_analysis.npy file to re-analyze
+    
+    save_iv : bool
+        Boolean that determines if the re-analyzed iva is saved
+        
+    save_bs : bool
+        Boolean that determines if the re-analyzed bsa is saved
+    """
+    iva2 = reanalyze_iv(iv_file, save_iv)
+    bsa2 = rerun_analysis(bs_file, iva2, save_bs)
+    
+    return iva2, bsa2
+
+
+###############################################################################
+# IVA translation functions below:
+# 1. Calculate TES parameters using thevenin equivalent voltage.
+# 2. Compute RP curve and storing parameters that determine RP curve.
+# 3. Save updated IVAnalysis object as a numpy file.
+###############################################################################
+
+def model_logalpha(r, logp0=0., p1=7., p2=1., logp3=0.):
+    """
+    Fits Alpha' parameter. P0 has units of [uW]^-1.
+    """
+    return logp0 + p1 * (1-r) + p2 * r ** np.exp(logp3) * np.log(1-r)
+
+
+def calculate_rpcurve(iva, minrfrac=0.2, maxrfrac=0.9):
+    """
+    Fit RP curve from IVA data. minrfrac and maxrfrac define the range
+    in which the RP curve is fit over.
+    
+    Args:
+    ----
+    iva : IVAnalysis
+    
+    
+    Returns:
+    --------
+    popts : np.array of shape (ndets x 6)
+        Fitted parameters for logp0, p1, p2, logp3, P_opt, and drn
+        P0 is optical power
+    """
+    nchans = iva['nchans']
+    popt_size = 6
+    popts = np.full((nchans, popt_size), np.nan)
+    
+    r1 = np.linspace(min(minrfrac, 0.0001), max(maxrfrac, 0.9999), 9999)
+    dr = r1[1] - r1[0]
+
+    def model_Pj(r, logp0=0., p1=7., p2=1., logp3=1.,
```

Comment:
Can you document here the mathematical description of the model function (or reference satoru's page), and also document the params?